### PR TITLE
Crud cash

### PR DESCRIPTION
**Pull request by Rustamxon7**
**CRUD Cash and CRUD Notes**

All requirements are complete:

- Create Cash and Notes that belong to the person ✅
- Add API route to fetch and display cash and notes data ✅

Render JSON formats for displaying data:
>     1. Include necessary data: in the parent index
>     2. Write a method to find the latest, the total for funds
>     3. Sort JSON data with methods

- Test categories validity ✅
- Solve the n+1 problem for users and categories ✅

Production link: [personal-budget-ui](https://personal-budget-ui.netlify.app/)
_For testing: Just in 💼...   😉
Email: test@test.com
Password: mmmmmm_

_Time spent: 6 days ⌛️_

Total spending time on 🗓

1. Authentification and Authorization: **3 days** (_including learning how to control JWT authentication with react!_)
2. CRUD Person:  4hs
3. CRUD Category: 2 days (_including implementing new styling based on my imagination_)
4. CRUD Cash and Note: 2 days (_including creating new methods for latest Transactions and solving n+1 problem_)

_Feel free to review my project and let me know if any changes need to be made. 😊_

### CODE REVIEW

File: app/controllers/api/v1/people_controller.rb [hunk 1]
```diff
@@ -3,18 +3,19 @@ class Api::V1::PeopleController < ApplicationController
...
-    render json: @people
+    @people = current_user.people.all.order(:name)
+    @categories = current_user.categories.all.order(:name)
+    render json: @people.as_json(include: :categories)
   end
 
   # GET /people/1
   def show
-    render json: @person
+    @person = Person.find(params[:id])
```
Comment (artyomlagun) [inline app/controllers/api/v1/people_controller.rb:13]: why did you remove render and left @person? you have :set_person before_action for it

File: app/models/fund.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+class Fund < ApplicationRecord
+  belongs_to :category
+
+  validates :title, presence: true, length: { maximum: 150 }
+  validates :note, presence: true, length: { maximum: 255 }
+
+  def last_funds
```
Comment (artyomlagun) [inline app/models/fund.rb:7]: I think it's better to use a scopes instead of methods

Comment (artyomlagun) [reply app/models/fund.rb:7]: also you have repeated `.order(date: :desc).limit(10)` - we should follow DRY principe

File: app/models/person.rb [hunk 3]
```diff
@@ -2,4 +2,16 @@ class Person < ApplicationRecord
...
   validates :name, presence: true, length: { maximum: 50 }, uniqueness: { scope: :user_id }
   validates :icon, presence: true
+
+  has_many :categories, dependent: :destroy
+
+  after_save :update_categories
+
+  private
+
+  def update_categories
```
Comment (artyomlagun) [inline app/models/person.rb:12]: for what did you create this?

File: app/controllers/api/v1/people_controller.rb (hunk 1 above)
Comment (Rustamxon7) [reply app/controllers/api/v1/people_controller.rb:13]: This error was made by mistake. Now it's fixed

File: app/models/fund.rb (hunk 2 above)
Comment (Rustamxon7) [reply app/models/fund.rb]: I have changed methods to scopes

File: app/models/person.rb (hunk 3 above)
Comment (Rustamxon7) [reply app/models/person.rb]: That one also made by mistake, I was checking it if works with API but it didn't made any impact to the project.

**Review by artyomlagun — APPROVED** [review]
